src/modules/dashboard/staff/components/staff-form.vue

```diff
@@ -52,51 +55,77 @@ const props = defineProps({
   },
 });
 
+const formSchema = toTypedSchema(
+  z.object({
+    name: z.string().min(1, i18n.global.t('Required field')).max(50),
+    lastname: z
+      .string()
+      .min(1, i18n.global.t('Required field'))
+      .max(50),
+    email: z
+      .string()
+      .min(1, i18n.global.t('Required field'))
+      .email(i18n.global.t('Invalid email')),
+    phoneNumber: z
+      .string()
+      .min(1, i18n.global.t('Required field'))
+      .length(9, i18n.global.t('Phone number must be 9 digits'))
+      .regex(/^\d+$/, i18n.global.t('Only digits are allowed')),
+  }),
+);
+
 const staffStore = useStaffStore();
-const staffData = ref(new StaffData());
-const errors = ref<Record<string, string>>({});
+const { handleSubmit, errors, resetForm } = useForm({
+  validationSchema: formSchema,
+  initialValues: getFormInitialValues(),
+});
 
-onMounted(() => {
+function getFormInitialValues() {
   if (props.actionType === ActionType.EDIT) {
     const member = staffStore.getMember(props.staffId);
     if (member) {
-      staffData.value.setData(member);
+      return {
+        name: member.name,
+        lastname: member.lastname,
+        email: member.email,
+        phoneNumber: member.phone,
+        role: member.role || Roles.ASSISTANT,
+      };
     }
   }
-});
+  return {
+    name: '',
+    lastname: '',
+    email: '',
+    phoneNumber: '',
+    role: Roles.ASSISTANT,
+  };
```

Comment:
To mozna wynieść do consta i nazwać `DEFAULT_MEMBER_DATA`

Reply:
Zrobione. Myślałem o tym wcześniej, ale tego nie zrobiłem bo korzystam z resetForma, którego wyciągam z useForm(). Gdybym miał swoją funkcję do resetowania to mógłbym tam używać tego obiektu, więc zamiast duplikowaniazwracałbym po prostu consta initialValues czy coś tam.